### DISCUSSION TITLE
Questions about service discovery for milvus 2.0

### DISCUSSION THREAD

**Discussion by lvmnn** [milvus-io/milvus#9999] [Q&A and General discussion]
Why do coord nodes need to use k8s service for service discovery, node nodes use etcd for service discovery.
why can't we use etcd for service discovery uniformly?

**Comment by godchen0212** [comment]
We are now using etcd to do service discovery, including coord nodes. It is possible that some desprecated codes has not been deleted which caused this misunderstanding.

**Reply by lvmnn** [reply]
Maybe I didn't express it clearly.
Why does the coord node read the virtual ip or domain name of the configuration file to register to etcd, and use k8s service for service discovery? Why not just read the local pod ip to register like  node nodes?

**Reply by yhmo** [reply]
In milvus, there are 3 ways to do service discovery:
1. read service address/port from environment variables
2. read service address/port from yaml configurations
3. register service address/port into etcd

The priority of the 3 ways is different. Ideally, each coord/node should follow same priority rule, but currently the rule is not clear and cause chaos. 

I thinks this is an enhancement issue and create one: https://github.com/milvus-io/milvus/issues/10093
